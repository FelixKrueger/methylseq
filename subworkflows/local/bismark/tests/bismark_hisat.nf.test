nextflow_workflow {

    name "Test Workflow BISMARK_HISAT"
    script "../main.nf"
    workflow "BISMARK"
    config "../nextflow.config"

    setup {
        run("UNTAR") {
            script "../../../../modules/nf-core/untar/main.nf"
            process {
                """
                input[0] = [
                    [:],
                    file(params.pipelines_testdata_base_path + 'reference/Hisat2_Index.tar.gz', checkIfExists: true)
                ]
                """
            }
        }
    }

    test("Bismark_Hisat | save_reference") {

        when {
            params {
                aligner              = "bismark_hisat"
                publish_dir_mode     = 'copy'
                unmapped             = false
                save_align_intermeds = false
                skip_deduplication   = false
                cytosine_report      = false
                rrbs                 = false
                nomeseq              = false
                slamseq              = false
                save_reference       = true
                outdir               = "results"
            }
            workflow {
                """

                input[0] = Channel.of([
                            [ id:'test', single_end:true ],
                            file(params.pipelines_testdata_base_path + 'testdata/SRR389222_sub1.fastq.gz', checkIfExists: true)
                ])
                input[1] = Channel.of([
                            [:],
                            file(params.pipelines_testdata_base_path + 'reference/genome.fa', checkIfExists: true)
                ])
                input[2] = UNTAR.out.untar
                input[3] = params.skip_deduplication
                input[4] = params.cytosine_report
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.bam.collect { bam(it[1]).getReadsMD5() },
                    workflow.out.dedup.collect { bam(it[1]).getReadsMD5() },
                    //workflow.out.mqc,
                    workflow.out.versions
                    ).match() }
            )
        }

    }

}
