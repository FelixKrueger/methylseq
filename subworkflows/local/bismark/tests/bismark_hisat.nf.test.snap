{
    "Bismark_Hisat: save_reference": {
        "content": [
            [
                "3a6485d719fcaf868c7f95c8f10a3f3c"
            ],
            [
                "dae445da532c973dc9149f3ceba2d2c5"
            ],
            [
                "versions.yml:md5,37c90312d5685ae87bcb080e0cab48de",
                "versions.yml:md5,66f7b51aeee12e293634cb74fea2f091",
                "versions.yml:md5,701e44c4be43ad049c16c7fc8686a21d",
                "versions.yml:md5,75082556508926250285e70e1932fa74",
                "versions.yml:md5,bff10f08bf0238c3303ef33749b7a93c",
                "versions.yml:md5,c06f3e25000330e948f7a35708270f3f",
                "versions.yml:md5,c2ad6c469b4d15c963f03b2228376a80",
                "versions.yml:md5,cb4a7015750ef1e0e504c56acb72a799"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-29T14:19:18.719515297"
    }
}